### PR TITLE
Refactor and add http resource

### PR DESCRIPTION
**Pull request by hckuo2**
@tsegismont 
I factor out methods for different purposes to make it more reusable.

### CODE REVIEW

**Comment by tsegismont** [comment]
Thanks @auszone

File: src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java [hunk 1]
```diff
@@ -120,72 +130,41 @@ public InventoryReporter(Vertx vertx, VertxHawkularOptions options, Context cont
...
+
+      reportFeed(fut1);
+      fut1.compose(aVoid1 -> {
+        reportRootResource(fut2);
+      }, fut2);
+      fut2.setHandler(aVoid1 -> {
+        Future<Void> rfut1 = Future.future();
+        Future<Void> rfut2 = Future.future();
+        reportEventbusResource(rfut1);
+        reportHttpServerResource(new SocketAddressImpl(8080, "0.0.0.0"), rfut2);
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java:152]: Next step is to detect when http servers are created (use the callbacks from the Metrics SPI).

Comment (hckuo2) [inline src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java:152]: In `public HttpServerMetrics<Long, Void, Void> createMetrics(HttpServer server, SocketAddress localAddress, HttpServerOptions options)` ?
But this method is called in a different thread how can i manage the order of the http requests?

Comment (tsegismont) [inline src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java:152]: You may create a map of existing servers at any point in time and then periodically inspect it. Or add a hook in Metrics impl to get notified when a http server is started/closed

Comment (hckuo2) [inline src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java:152]: solved 👍

**Pull request merged**